app/connectors/connector-verifier/src/main/java/io/syndesis/verifier/api/MetadataAdapter.java

```diff
@@ -33,11 +34,6 @@
      * {@link SyndesisMetadata}. Method will receive all properties that client
      * specified and the retrieved {@link MetaData} from the
      * appropriate Camel {@link MetaDataExtension}.
-     *
-     * @param properties properties specified on the endpoint
-     * @param metadata the retrieved metadata
-     * @return Syndesis styled metadata
      */
-    SyndesisMetadata adapt(String actionId, Map<String, Object> properties, MetaData metadata);
-
+    SyndesisMetadata fetch(CamelContext context, String componentId, String actionId, Map<String, Object> properties);
```

Comment:
Since this interface is `fetch`-ing, perhaps we should rename it to `MetadataRetrieval`